consensus/cryptoiface.go

```diff
@@ -0,0 +1,82 @@
+package consensus
+
+import (
+	"github.com/relab/hotstuff"
+)
+
+// CryptoBase provides the basic cryptographic methods needed to create, verify, and combine signatures.
+type CryptoBase interface {
```

Comment:
See the naming/composition here: [`Signer`](https://pkg.go.dev/crypto#Signer). There is no interface that includes a verify function. On the other hand, it seems the std lib doesn't define an interface that's used across signature types. Instead most signing/verify algorithms just use functions specific for each signature type.

Reply:
I guess my point was that I don't think `CryptoBase` is the right name for this interface, since the methods are related to digital signatures in some form. The other question is whether or not we should use an interface.

Reply:
Similar to other question: why is this in `consensus`?

Reply:
I haven't yet found a better place to put these types/interfaces without import cycles. It would be best to keep them in the crypto package, but I also need the interfaces to be available in the module system. And the crypto package needs the module system...

Reply:
We should ideally refactor things to avoid such strong coupling, but that should be done as a separate PR.

Reply:
Marking this as resolved since I created an issue for the possible refactor of the module system.